hpobench/benchmarks/mo/cnn_benchmark.py

```diff
@@ -0,0 +1,591 @@
+        rng : np.random.RandomState, int, None
+            Random seed to use in the benchmark.
+            To prevent overfitting on a single seed, it is possible to pass a
+            parameter ``rng`` as 'int' or 'np.random.RandomState' to this function.
+            If this parameter is not given, the default random state is used.
+        shuffle: bool, None
+            If ``True``, shuffle the training idx. If no parameter ``rng`` is given, use the class random state.
+            Defaults to ``False``.
+        kwargs
+
+        Returns
+        -------
+        Dict -
+            function_value : Dict
+                validation_accuracy: float
+                model_size: float
+            cost : time to train the network
+            info : Dict
+                train_accuracy : float,
+                training_cost : float,
+                valid_accuracy : float,
+                valid_cost : float,
+                test_accuracy : float,
+                test_cost : float,
+                model_size : int,
+                fidelity : Dict
+                    used fidelities in this evaluation
+        """
+        self.rng = rng_helper.get_rng(rng=rng, self_rng=self.rng)
+        self.__seed_everything()
+
+        if shuffle:
+            self._shuffle_data(rng=self.rng, shuffle_valid=False)
+
+        time_in = time.time()
+
+        device = torch.device('cuda') if torch.cuda.is_available() else torch.device('cpu')
+        logger.info(f'We use the device: {device}')
+
+        # initializing model
+        model = self.init_model(configuration, rng).to(device)
+        epochs = fidelity['budget']
+
+        optimizer = torch.optim.Adam(model.parameters(), lr=configuration['learning_rate_init'])
+        criterion = torch.nn.CrossEntropyLoss()
+
+        ds_train = TensorDataset(self.X_train, self.y_train)
+        ds_train = DataLoader(ds_train, batch_size=configuration['batch_size'], shuffle=True)
+
+        ds_val = TensorDataset(self.X_valid, self.y_valid)
+        ds_val = DataLoader(ds_val, batch_size=configuration['batch_size'], shuffle=True)
+
+        ds_test = TensorDataset(self.X_test, self.y_test)
+        ds_test = DataLoader(ds_test, batch_size=configuration['batch_size'], shuffle=True)
+
+        start = time.time()
+        t = tqdm.tqdm(total=epochs)
+
+        train_accuracy = 0
+        for epoch in range(epochs):
+            train_accuracy = model.train_fn(optimizer, criterion, ds_train, device).item()
+            t.set_postfix(train_accuracy=train_accuracy)
+            t.update()
+        training_runtime = time.time() - start
+
+        num_params = np.sum([p.numel() for p in model.parameters()]).item()
+        start = time.time()
+        val_accuracy = model.eval_fn(ds_val, device).item()
+        eval_valid_runtime = time.time() - start
+        start = time.time()
+        test_accuracy = model.eval_fn(ds_test, device).item()
```

Comment:
Same questions as for the other benchmark: Why spending time on computing test metrics?

Reply:
Good question. Changed it to "training time". 

The eval time should be almost equal for every run, so i think this is nothing of interest.